coala_quickstart/generation/SettingsFilling.py

```diff
@@ -165,3 +168,93 @@ def resolve_anomaly(setting_name,
                                                    join_names(values)),
                   REPORT_ANOMALY_COLOR))
     return input()
+
+
+def require_setting(setting_name, setting_info, section):
+    """
+    This method is responsible for prompting a user about a missing setting and
+    taking its value as input from the user.
+
+    :param setting_name: Name of the setting missing
+    :param setting_info: A dictionary of the form
+
+    ::
+        {
+            "help_text": "help string for the setting",
+            "bears": [SomeBear", "SomeOtherBear"],
+            "type": bool
+        }
+
+    :param section:      The section the action corresponds to.
+    :return:             The preferred value provided by the user
+                         for the setting.
+    """
+    needed = join_names(setting_info["bears"])
+
+    STR_GET_VAL_FOR_SETTING = ('\nPlease enter a value for the setting \"{}\" '
+                               '({}) needed by {} for section \"{}\": ')
+
+    STR_REPORT_INVALID_VALUE_TYPE = ('coala-quickstart was unable to convert '
+                                     'your input to the type {} required for '
+                                     'the setting.')
+
+    REQUIRED_SETTINGS_COLOR = 'green'
+    REPORT_INVALID_TYPE_COLOR = 'cyan'
+
+    print(colored(STR_GET_VAL_FOR_SETTING.format(setting_name,
+                                                 setting_info["help_text"],
+                                                 needed,
+                                                 section.name),
+                  REQUIRED_SETTINGS_COLOR))
+
+    user_input = input()
+
+    if setting_info["type"]:
+        try:
+            user_input = setting_info["type"](user_input)
+        except ValueError:
+            print(colored(
+                STR_REPORT_INVALID_VALUE_TYPE.format(setting_info["type"]),
+                REPORT_INVALID_TYPE_COLOR))
+            return require_setting(setting_name, setting_info, section)
```

Comment:
I really hope the user doesn't make the same mistake for a thousand times (999 is the recursion limit in Python IIRC) :P